include/seqan3/io/alignment_file/detail.hpp

```diff
@@ -574,37 +579,37 @@ template <TupleLike alignment_type>
     requires std::tuple_size_v<remove_cvref_t<alignment_type>> == 2 &&
              detail::all_satisfy_aligned_seq<detail::tuple_type_list_t<alignment_type>>
 //!\endcond
-void alignment_from_cigar(alignment_type & alignment, std::vector<std::pair<char, size_t>> const & cigar)
+void alignment_from_cigar(alignment_type & alignment, std::vector<cigar> const & cigar_vector)
 {
     using std::get;
     auto current_ref_pos  = std::ranges::begin(get<0>(alignment));
     auto current_read_pos = std::ranges::begin(get<1>(alignment));
 
-    for (auto [cigar_op, cigar_count] : cigar)
+    for (auto [cigar_count, op] : cigar_vector)
     {
-        assert(is_char<'M'>(cigar_op) || is_char<'='>(cigar_op) || is_char<'X'>(cigar_op) || is_char<'D'>(cigar_op) ||
-               is_char<'N'>(cigar_op) || is_char<'I'>(cigar_op) || is_char<'P'>(cigar_op)); // checked during IO
+        assert(('M'_cigar_op == op) || ('='_cigar_op == op) || ('X'_cigar_op == op) || ('D'_cigar_op == op) ||
+               ('N'_cigar_op == op) || ('I'_cigar_op == op) || ('P'_cigar_op == op)); // checked during IO
 
-        if (is_char<'M'>(cigar_op) || is_char<'='>(cigar_op) || is_char<'X'>(cigar_op))
+        if (('M'_cigar_op == op) || ('='_cigar_op == op) || ('X'_cigar_op == op))
```

Comment:
This can be replaced with a `switch(op)` statement.

Reply:
I did replace it in when the code was moved in to the sam format